src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs

```diff
@@ -323,6 +323,38 @@ public void WarningLevel_2()
             }
         }
 
+        [Fact]
+        public void NullableWarnings()
+        {
+            foreach (ErrorCode error in Enum.GetValues(typeof(ErrorCode)))
+            {
+                if ((int)error < 8600 || (int)error >= 9000)
+                {
+                    continue;
+                }
+
+                if (!error.ToString().StartsWith("WRN"))
+                {
+                    // Only interested in warnings
+                    continue;
+                }
+
+                if (ErrorFacts.NullableWarnings.Contains(MessageProvider.Instance.GetIdForErrorCode((int)error)))
+                {
+                    continue;
+                }
+
+                // Nullable-unrelated warnings in the C# 8 range should be added to this array.
+                var nullableUnrelatedWarnings = new[]
```

Comment:
We won't be adding new warnings in C# 8 so this validates the existing code but doesn't really protect against the future, where we add warnings in future language versions.

Reply:
Yeah, let's consider how we want to handle this moving forward (it seems like we wouldn't expect most of the warnings in C# 9 to be nullable-related?)